src/rpcblockchain.cpp

```diff
@@ -1141,7 +1143,14 @@ bool AdvertiseBeacon(std::string &sOutPrivKey, std::string &sOutPubKey, std::str
                 sError = "ALREADY_IN_CHAIN";
                 return false;
             }
-            
+
+            // Prevent users from advertising multiple times in succession by setting a limit of one advertisement per 5 blocks.
+            // Realistically 1 should be enough however just to be sure we deny advertisements for 5 blocks.
+            if ((nBestHeight - nLastBeaconAdvertised) < 5)
```

Comment:
You can remove the global variable and make it static under the comment: `static int nLastBeaconAdvertised = 0`.